src/Draco.RedGreenTree.SourceGeneration/IncrementalGeneratorInitializationContextExtensions.cs

```diff
@@ -0,0 +1,25 @@
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Draco.RedGreenTree.SourceGeneration;
+
+internal static class IncrementalGeneratorInitializationContextExtensions
+{
+    public static IncrementalValuesProvider<AttributeAndType> ForTypesWithAttribute(this IncrementalGeneratorInitializationContext ctx, string attributeName) =>
+        ctx.SyntaxProvider.ForAttributeWithMetadataName(
+            attributeName,
+            (syntaxNode, ct) => syntaxNode is TypeDeclarationSyntax,
+            AttributeAndType? (ctx, ct) =>
```

Comment:
I personally tend to cast the null to `AttributeAndType?` in this case and not specify a return type for lambdas, but this is just my stylistic thing, it can stay